bin/collator/src/cli.rs

```diff
@@ -74,56 +71,6 @@ pub enum Subcommand {
     TryRuntime(try_runtime_cli::TryRuntimeCmd),
 }
 
-fn validate_relay_chain_url(arg: &str) -> Result<Url, String> {
-    let url = Url::parse(arg).map_err(|e| e.to_string())?;
-
-    if url.scheme() == "ws" {
-        Ok(url)
-    } else {
-        Err(format!(
-            "'{}' URL scheme not supported. Only websocket RPC is currently supported",
-            url.scheme()
-        ))
-    }
-}
-
-#[allow(missing_docs)]
-#[derive(Debug, clap::Parser)]
-pub struct RunCmd {
```

Comment:
No need to have parachain id as an RunCmd arg, since dev chains (astar-dev, shiden-dev, shibuya-dev) run with their corresponding non-dev chains' parachain ids (astar, shiden, shibuya)

Even taking `--parachain-id`, dev chains cannot produce blocks when connected to Relaychains since their genesis state is pre-defined with fixed number (it was `2000`) in ChainSpec. custom arg `--parachain-id` has no convenience rather confusing.

Reply:
I don't necessarily agree with the first point's explanation because it seems that the problem being solved has been turned upside-down. I agree with the point you're trying to make though.

The reason why blocks cannot be produced and `2000` is used as para-id is because `build-spec` command doesn't take *parachainId* as an argument. But then again, even if it did take paraId as argument, having this argument again as part of collator starting CLI is redundant.

Good call to remove it!